MapboxNavigation/Abbreviations.swift

```diff
@@ -0,0 +1,129 @@
+import UIKit
+
+// http://wiki.openstreetmap.org/wiki/Name_finder:Abbreviations#English
+let Abbreviations = [
+    "apartments": "Apts",
+    "center": "Ctr",
+    "centre": "Ctr",
+    "county": "Co",
+    "creek": "Crk",
+    "crossing": "Xing",
+    "downtown": "Dtwn",
+    "father": "Fr",
+    "fort": "Ft",
+    "heights": "Hts",
+    "international": "Int’l",
+    "junction": "Jct",
+    "junior": "Jr",
+    "lake": "Lk",
+    "market": "Mkt",
+    "memorial": "Mem",
+    "mount": "Mt",
+    "mountain": "Mtn",
+    "national": "Nat’l",
+    "park": "Pk",
+    "point": "Pt",
+    "river": "Riv",
+    "route": "Rte",
+    "saint": "St",
+    "saints": "SS",
+    "school": "Sch",
+    "senior": "Sr",
+    "sister": "Sr",
+    "square": "Sq",
+    "station": "Sta",
+    "township": "Twp",
+    "university": "Univ",
+    "village": "Vil",
+    "william": "Wm",
+]
+
+let CompassDirections = [
+    "east": "E",
+    "north": "N",
+    "northeast": "NE",
+    "northwest": "NW",
+    "south": "S",
+    "southeast": "SE",
+    "southwest": "SW",
+    "west": "W",
+]
+
+let Classifications = [
+    "alley": "Aly",
+    "avenue": "Ave",
+    "boulevard": "Blvd",
+    "bridge": "Br",
+    "bypass": "Byp",
+    "circle": "Cir",
+    "court": "Ct",
+    "cove": "Cv",
+    "crescent": "Cres",
+    "drive": "Dr",
+    "expressway": "Expy",
+    "freeway": "Fwy",
+    "footway": "Ftwy",
+    "highway": "Hwy",
+    "lane": "Ln",
+    "motorway": "Mwy",
+    "parkway": "Pky",
+    "plaza": "Plz",
+    "pike": "Pk",
+    "point": "Pt",
+    "place": "Pl",
+    "road": "Rd",
+    "square": "Sq",
+    "street": "St",
+    "terrace": "Ter",
+    "turnpike": "Tpk",
+    "walk": "Wk",
+    "walkway": "Wky",
+]
+
+/// Options that specify what kinds of words in a string should be abbreviated.
+struct StringAbbreviationOptions : OptionSet {
+    let rawValue: Int
+    
+    /// Abbreviates ordinary words that have common abbreviations.
+    static let Abbreviations = StringAbbreviationOptions(rawValue: 1 << 0)
+    /// Abbreviates directional words.
+    static let Directions = StringAbbreviationOptions(rawValue: 1 << 1)
+    /// Abbreviates road name suffixes.
+    static let Classifications = StringAbbreviationOptions(rawValue: 1 << 2)
+}
+
+extension String {
+    /// Returns an abbreviated copy of the string.
+    func stringByAbbreviatingWithOptions(options: StringAbbreviationOptions) -> String {
```

Comment:
Swift 3 has different method naming conventions than Swift 2: let’s call this method `abbreviated(by:)`.